bindings/java/c_kzg_4844_jni.c

```diff
@@ -146,151 +179,170 @@ JNIEXPORT jbyteArray JNICALL Java_ethereum_ckzg4844_CKZG4844JNI_computeKzgProof(
-  jbyteArray commitment = (*env)->NewByteArray(env, BYTES_PER_COMMITMENT);
-  KZGCommitment *commitment_native = (KZGCommitment *)(*env)->GetByteArrayElements(env, commitment, NULL);
+  size_t commitment_bytes_size = (size_t)(*env)->GetArrayLength(env, commitment_bytes);
+  if (commitment_bytes_size != BYTES_PER_COMMITMENT)
+  {
+    throw_invalid_size_exception(env, "Invalid commitment size.", commitment_bytes_size, BYTES_PER_COMMITMENT);
+    return 0;
+  }
 
-  C_KZG_RET ret = blob_to_kzg_commitment(commitment_native, (const Blob *)blob_native, settings);
+  size_t proof_bytes_size = (size_t)(*env)->GetArrayLength(env, proof_bytes);
+  if (proof_bytes_size != BYTES_PER_COMMITMENT)
+  {
+    throw_invalid_size_exception(env, "Invalid proof size.", proof_bytes_size, BYTES_PER_PROOF);
+    return 0;
+  }
 
-  (*env)->ReleaseByteArrayElements(env, blob, blob_native, JNI_ABORT);
-  (*env)->ReleaseByteArrayElements(env, commitment, (jbyte *)commitment_native, 0);
+  Blob *blob_native = (Blob *)(*env)->GetByteArrayElements(env, blob, NULL);
+  Bytes48 *commitment_native = (Bytes48 *)(*env)->GetByteArrayElements(env, commitment_bytes, NULL);
+  Bytes48 *proof_native = (Bytes48 *)(*env)->GetByteArrayElements(env, proof_bytes, NULL);
+
+  bool out;
+  C_KZG_RET ret = verify_blob_kzg_proof(&out, blob_native, commitment_native, proof_native, settings);
+
+  (*env)->ReleaseByteArrayElements(env, blob, (jbyte *)blob_native, JNI_ABORT);
+  (*env)->ReleaseByteArrayElements(env, commitment_bytes, (jbyte *)commitment_native, JNI_ABORT);
+  (*env)->ReleaseByteArrayElements(env, proof_bytes, (jbyte *)proof_native, JNI_ABORT);
 
   if (ret != C_KZG_OK)
   {
-    throw_c_kzg_exception(env, ret, "There was an error while converting blob to commitment.");
-    return NULL;
+    throw_c_kzg_exception(env, ret, "There was an error in verifyBlobKzgProof.");
+    return 0;
   }
 
-  return commitment;
+  return (jboolean)out;
 }
 
-JNIEXPORT jboolean JNICALL Java_ethereum_ckzg4844_CKZG4844JNI_verifyKzgProof(JNIEnv *env, jclass thisCls, jbyteArray commitment_bytes, jbyteArray z_bytes, jbyteArray y_bytes, jbyteArray proof_bytes)
+JNIEXPORT jboolean JNICALL Java_ethereum_ckzg4844_CKZG4844JNI_verifyBlobKzgProofBatch(JNIEnv *env, jclass thisCls, jbyteArray blobs, jbyteArray commitments_bytes, jbyteArray proofs_bytes, jlong count)
 {
   if (settings == NULL)
   {
     throw_exception(env, TRUSTED_SETUP_NOT_LOADED);
     return 0;
   }
 
-  Bytes48 *commitment_native = (Bytes48 *)(*env)->GetByteArrayElements(env, commitment_bytes, NULL);
-  Bytes48 *proof_native = (Bytes48 *)(*env)->GetByteArrayElements(env, proof_bytes, NULL);
-  Bytes32 *z_native = (Bytes32 *)(*env)->GetByteArrayElements(env, z_bytes, NULL);
-  Bytes32 *y_native = (Bytes32 *)(*env)->GetByteArrayElements(env, y_bytes, NULL);
+  size_t count_native = (size_t)count;
+  size_t blobs_size = (size_t)(*env)->GetArrayLength(env, blobs);
+  if (blobs_size != count_native * BYTES_PER_BLOB)
```

Comment:
Similar to above, I think we should panic horribly if the inputs are not the right size, and not do any attempt at failing gracefully. I don't see why a caller would ever give us a wrong-sized blob or commitment.

All these checks can be compressed into three lines of code.

Reply:
I really disagree with the idea of "panic horribly" with the bindings. A little extra code that handles things properly is the way to go IMO. @StefanBratanov what are your thoughts on this?

Reply:
Hi, my preference is to have concise exceptions in case things go bad. Also I think it's cleaner (even though more lines) since you can see exactly what exception will be thrown with each check. Also not sure just using assert is going to work and will propagate the exception correctly to java.